### PR TITLE
PHP 7.0 Scalar Type Hinting Support (Should be BC)

### PR DESCRIPTION
**Pull request by jmcclell**

### CODE REVIEW

**Comment by jmcclell** [comment]
Oof. Any guess on that stack overflow in HHVM? Things seem normal to me otherwise. /sigh

File: tests/CG/Tests/Generator/Fixture/generated/php7_class.php [hunk 1]
```diff
@@ -0,0 +1,61 @@
+namespace CG\Tests\Generator\Fixture;
```
Comment (stof) [inline tests/CG/Tests/Generator/Fixture/generated/php7_class.php]: aren't you missing the php opening tag in the file ?

Comment (jmcclell) [inline tests/CG/Tests/Generator/Fixture/generated/php7_class.php]: From what I could tell, none of the generated files had opening PHP tags. It seems that this library inserts the opening PHP tag in the generator itself - the Enhancer class, for example, seems to make the assumption that it has to insert the start tag. Am I making a fundamental mistake in my understanding of your structure?

**Comment by perk11** [comment]
Could this get merged? Lack of this is making JMS not compatible with PHP7

**Pull request merged**